README.rst

```diff
@@ -308,6 +308,122 @@ In case you need to run additional instructions just before the build step you c
 
 You can find more patches in the `patch catalog <#template-patch-catalog>`_ below.
 
+Using Frontend Plugin Slots
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``mfe-dockerfile-pre-npm-build`` patch is also suitable for taking advantage of the ``env.config.jsx`` mechanism, which lets you define and use frontend plugins.  For instance, you can use it to modify the footer across all MFEs *without* having to maintain a fork of ``frontend-component-footer``.
+
+Start by following the Tutor plugin tutorial at the `Adding new templates <https://docs.tutor.edly.io/tutorials/plugin.html#adding-new-templates>`_ section.  Instead of adding a whole new ``Dockerfile``, though, you'll just modify the exiting one.  Your ``myplugin.py`` should look something like this:
+
+.. code-block:: python
+
+    import os
+    from tutor import hooks
+
+    # Add the template root folder
+    template_folder = os.path.join(os.path.dirname(__file__), "templates")
+    hooks.Filters.ENV_TEMPLATE_ROOTS.add_item(template_folder)
+
+    # env.config.jsx files should live in `mfe/build/mfe`, which is also where the
+    # tutor-mfe Dockerfile goes.  This is so it's more straightforward to copy it
+    # in, as seen below.
```

Comment:
I realize it's iffy to recommend a plugin put a file directly into one of tutor-mfe's own ones, but for some reason I couldn't get the t utor-mfe Dockerfile to COPY from a relative parent.

Reply:
In [Indigo](https://github.com/overhangio/tutor-indigo/blob/master/tutorindigo/plugin.py#L50), I implemented a similar setup, but I have some concerns regarding the inclusion of `env.config.jsx` in mfe/build/mfe. Specifically, if we go this route, each user will need to take similar steps as I did in Indigo. This means they’ll either have to fork tutor-mfe and modify `env.config.jsx` or copy `env.config.jsx` from their custom plugin into `mfe/build/mfe`.

Can we think of any workaround to update or change `env.config.jsx` more seamlessly without requiring each user to fork plugin or override this file? 

OR I am unsure that this `env.config.jsx` file will raise any issue for[ indigo plugin](https://github.com/overhangio/tutor-indigo/blob/master/tutorindigo/plugin.py#L134).

Reply:
Ah, I didn't look at how indigo does it.  Let me check.

But yeah, the fact multiple plugins may want to modify env.config.jsx is not something I took into account, here.